File: day4.py
import hashlib

d5_input = "iwrupvqb"


def find_with_prefix(prefix: str) -> int:
    curr = 1

    while True:
        md = hashlib.md5()
        md.update(f"{d5_input}{curr}".encode("ascii"))
        h = md.hexdigest()
        if h.startswith(prefix):
            break
        else:
            curr += 1

    return curr


print(f"Part 1: {find_with_prefix('00000')}")
print(f"Part 2: {find_with_prefix('000000')}")

File: src/day4.jl
using MD5: md5

d5_input = "iwrupvqb"

function find_with_prefix(prefix::String)::Int
    curr = 1
    while true
        h = bytes2hex(md5("$d5_input$curr"))
        if startswith(h, prefix)
            break
        else
            curr += 1
        end
    end
    curr
end

part_1 = find_with_prefix("00000")
println("Part 1: $part_1")
part_2 = find_with_prefix("000000")
println("Part 2: $part_2")

